api/base/permissions.py

```diff
@@ -0,0 +1,14 @@
+def PermissionWithGetter(Base, getter):
+    """A psuedo class for checking permissions
+    of subresources without having to redefine permission classes
+    """
+    class Perm(Base):
+        def get_object(self, request, view, obj):
+            if callable(getter):
+                return getter(request, view, obj)
+            return getattr(obj, getter)
```

Comment:
We could allow getting nested attributes ,e.g. `sanction.node` , if you use `operator.attrgetter`, but it's not necessary at the moment.